### PR TITLE
[MI-3241] : Added delete config modal to the webapp

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
Developed and added delete config modal to delete the existing configuration of plugin

### CODE REVIEW

File: webapp/src/containers/components/modals/deleteConfigModal.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React, {useEffect, useState} from 'react';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import './styles.css';
+
+interface Props {
+    visible: boolean;
+    setVis: React.Dispatch<React.SetStateAction<boolean>>;
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/deleteConfigModal.tsx]:
```suggestion
    setVisibility: React.Dispatch<React.SetStateAction<boolean>>;
```

File: webapp/src/containers/components/modals/deleteConfigModal.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                        onClick={handleClose}
+                    >{'Close'}</Button>
+                    <Button variant='danger'>{'Delete'}</Button>
+                </Modal.Footer>
+            </Modal>
+        </>
+    );
+}
+
+export default DeleteConfigModal;
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/deleteConfigModal.tsx]: Add EOF

File: webapp/src/types/mattermostWebapp/index.d.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 export interface PluginRegistry {
     registerRootComponent(component: ReactDOM)
-    registerAdminConsoleCustomSetting(key: string, component: React.ElementType)
+    registerAdminConsoleCustomSetting(key: string, component: React.ElementType);
```
Comment (raghavaggarwal2308) [inline webapp/src/types/mattermostWebapp/index.d.ts]: why this change?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/types/mattermostWebapp/index.d.ts (hunk 3 above)
Comment (Kshitij-Katiyar) [reply webapp/src/types/mattermostWebapp/index.d.ts]: reverted

File: webapp/src/containers/components/modals/deleteConfigModal.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import React, {useEffect, useState} from 'react';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import './styles.css';
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/deleteConfigModal.tsx]:
```suggestion

import './styles.css';
```

File: webapp/src/containers/components/modals/deleteConfigModal.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+interface Props {
+    visible: boolean;
+    setVisibility: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+function DeleteConfigModal(props: Props) {
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/deleteConfigModal.tsx]: use `types` and arrow function everywhere

File: webapp/src/containers/components/modals/deleteConfigModal.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    useEffect(() => {
+        setShow(props.visible);
+    }, [props.visible]);
+
+    const handleClose = () => {
+        setShow(false);
+        props.setVisibility(false);
+    };
+    return (
+        <>
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/deleteConfigModal.tsx]: why you are using this?

File: webapp/src/containers/components/modals/deleteConfigModal.tsx (hunk 2 above)
Comment (avas27JTG) [reply webapp/src/containers/components/modals/deleteConfigModal.tsx]: +1

File: webapp/src/containers/components/modals/deleteConfigModal.tsx (hunk 5 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/deleteConfigModal.tsx]: i explored the difference between them, any particular reason to use arrow function

File: webapp/src/containers/components/modals/deleteConfigModal.tsx (hunk 6 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/deleteConfigModal.tsx]: removed the <></> since we have only one component in file

Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/deleteConfigModal.tsx]: didn't get what you are suggesting on lines 46-49

File: webapp/src/containers/components/modals/deleteConfigModal.tsx (hunk 5 above)
Comment (avas27JTG) [reply webapp/src/containers/components/modals/deleteConfigModal.tsx]: Arrow functions are preferred for their concise syntax, lexical `this` binding, and implicit return which are important.
You can explore in detail coding standards for ECMAScript.

Comment (avas27JTG) [reply webapp/src/containers/components/modals/deleteConfigModal.tsx]: also `interface` is used where there is a need to extend or override props like in libraries but for custom components or components local to a project where props of one component are not going to be used another then it's advised to use `types`.
make these changes.

File: webapp/src/containers/components/modals/deleteConfigModal.tsx (hunk 6 above)
Comment (avas27JTG) [reply webapp/src/containers/components/modals/deleteConfigModal.tsx]: You need to add an empty line at the end of every file, see the red icon below it means there is no new line
image

Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/deleteConfigModal.tsx]: ok

File: webapp/src/containers/components/modals/deleteConfigModal.tsx (hunk 5 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/deleteConfigModal.tsx]: okay, thanks

**Review by raghavaggarwal2308 — APPROVED** [review]

**Review by avas27JTG — APPROVED** [review]
